utils/mongo.py

```diff
@@ -21,54 +23,54 @@ def __init__(self, connection, document_name):
         self.logger = logging.getLogger(__name__)
 
     # <-- Pointer Methods -->
-    async def update(self, dict):
+    async def update(self, dict: Dict):
         """
         For simpler calls, points to self.update_by_id
         """
         await self.update_by_id(dict)
 
-    async def get_by_id(self, id):
+    async def get_by_id(self, _id: Any, *, projection: Union[Dict, None]=None):
         """
         This is essentially find_by_id so point to that
         """
-        return await self.find_by_id(id)
+        return await self.find_by_id(_id, projection=projection)
 
-    async def find(self, id):
+    async def find(self, _id: Any):
         """
         For simpler calls, points to self.find_by_id
         """
-        return await self.find_by_id(id)
+        return await self.find_by_id(_id)
 
-    async def delete(self, id):
+    async def delete(self, _id: Any):
         """
         For simpler calls, points to self.delete_by_id
         """
-        await self.delete_by_id(id)
+        await self.delete_by_id(_id)
 
     # <-- Actual Methods -->
-    async def find_by_id(self, id):
+    async def find_by_id(self, _id, *, projection: Union[Dict, None]=None):
         """
-        Returns the data found under `id`
+        Returns the data found under `_id`
         Params:
-         -  id () : The id to search for
+         -  _id () : The _id to search for
         Returns:
          - None if nothing is found
          - If somethings found, return that
         """
-        return await self.db.find_one({"_id": id})
+        return await self.db.find_one({"_id": _id}, projection or {})
```

Comment:
`projection` is very cool. It makes to shorter the query; similar to SQL
```sql
SELECT _id FROM collection WHERE _id=123;
```
```py
await collection.find_one({"_id": 123}, {"_id": 1})
```
If no projection is given
```sql
SELECT * FROM collection WHERE _id=123;
```
```py
await collection.find_one({"_id": 123})
```